### PR TITLE
VZ 5800: Detect Jaeger endpoint from Verrazzano managed namespace

### PR DESCRIPTION
**Pull request by anders-swanson**
The Jaeger collector endpoint should reside in a Verrazzano managed namespace.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/istio/jaeger.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		port := zipkinPort(*service)
+		collectorURL := fmt.Sprintf("%s.%s.svc.cluster.local:%d",
+			service.Name,
+			service.Namespace,
+			port,
+		)
+		return []vzapi.InstallArgs{
+			{
+				Name:  meshConfigTracingTLSMode,
+				Value: "ISTIO_MUTUAL",
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/istio/jaeger.go:46]: i cannot tell where ISTIO_MUTUAL is being used

File: platform-operator/controllers/verrazzano/component/istio/jaeger.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+				Value: collectorURL,
+			},
+		}, nil
+	}
+
+	return nil, nil
+}
+
+//findFirstJaegerCollectorService gets the first Jaeger collector service in the cluster that is not a headless service
+func findFirstJaegerCollectorService(ctx spi.ComponentContext) (*v1.Service, error) {
```
[the remark this answers is not in the document]
Comment (anders-swanson) [reply platform-operator/controllers/verrazzano/component/istio/jaeger.go:59]: Istio discovers the Jaeger collector service to configure the tracing endpoint

File: platform-operator/controllers/verrazzano/component/istio/jaeger.go (hunk 1 above)
Comment (anders-swanson) [reply platform-operator/controllers/verrazzano/component/istio/jaeger.go:46]: In this case, ISTIO_MUTUAL is used when Istio sends traces to Jaeger collector

File: platform-operator/controllers/verrazzano/component/istio/jaeger.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	services := &v1.ServiceList{}
+	selector := clipkg.MatchingLabels{
+		constants.KubernetesAppLabel: constants.JaegerCollectorService,
+	}
+	if err := ctx.Client().List(context.TODO(), services, selector); err != nil {
```
[the remark this answers is not in the document]
Comment (kminder) [reply platform-operator/controllers/verrazzano/component/istio/jaeger.go:64]: Can there be more than one Jaeger instance?  If so how do we know the correct one is being picked?  Can the user override this magic?

[the remark this answers is not in the document]
Comment (anders-swanson) [reply platform-operator/controllers/verrazzano/component/istio/jaeger.go:64]: updated to allow user overrides of zipkin.address

**Review by pfmackin — APPROVED** [review]

**Review by kminder — DISMISSED** [review]
Approved.

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**